libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -231,50 +232,89 @@ end
+                                             endpoint_LF[3]+ans_LF[3]+gait_pos_z[4])
+    angles_LF[1] = -135 + angles_LF[1]
     Gaitselect()
-    current = {angles1[1],angles1[2],angles1[3],angles2[1],angles2[2],angles2[3],angles3[1],angles3[2],angles3[3],angles4[1],angles4[2],angles4[3]}
+    current_angle = {angles_RF[1],angles_RF[2],angles_RF[3],
+                     angles_LF[1],angles_LF[2],angles_LF[3],
+                     angles_LB[1],angles_LB[2],angles_LB[3],
+                     angles_RB[1],angles_RB[2],angles_RB[3]}
 
     if servo_estimate() then
         start_time = millis()
         calc_gait_sequence()
-        last_angle = current
+        last_angle = current_angle
     end
 
-    return current
+    return current_angle
 end
 
 -- servo angles when robot is disarmed and resting body on the ground
 local rest_angles = { 45, -90, 40,      -- front right leg (coxa, femur, tibia)
                      -45, -90, 40,      -- front left leg (coxa, femur, tibia)
-                      45, -90, 40,      -- back right leg (coxa, femur, tibia)
-                     -45, -90, 40}      -- back left leg (coxa, femur, tibia)
-
-local servo_direction = { 1,  1,  1,    -- front right leg (coxa, femur, tibia)
-                         -1, -1, -1,    -- front left leg (coxa, femur, tibia)
-                         -1, -1,  1,    -- back right leg (coxa, femur, tibia)
-                          1,  1, -1}    -- back left leg (coxa, femur, tibia)
+                     -45, -90, 40,      -- back left leg (coxa, femur, tibia)
+                      45, -90, 40}      -- back right leg (coxa, femur, tibia)
 
 function update()
-    x_speed = vehicle:get_control_output(control_input_throttle) * xy_speed_max
-    yaw_speed = vehicle:get_control_output(control_input_yaw) * yaw_speed_max
+    throttle = vehicle:get_control_output(control_input_throttle) * xy_travel_max
+
+    if throttle > 0 then
+        gait_direction = -1
+        y_travel = throttle
+    elseif throttle < 0 then
+        gait_direction = 1
+        y_travel = - throttle
+    elseif throttle == 0 then
+        gait_direction = 1
+        y_travel = 0
+    end
 
+    yaw_travel = -vehicle:get_control_output(control_input_yaw) * yaw_lenght_max
     body_rot_x = -vehicle:get_control_output(control_input_roll) * body_rot_max
     body_rot_y = -vehicle:get_control_output(control_input_pitch) * body_rot_max
+    body_pos_z =  vehicle:get_control_output(control_input_height) * height_max
+
+    local servo_direction = { gait_direction *  1, -1,  1,    -- front right leg (coxa, femur, tibia)
+                              gait_direction *  1,  1, -1,    -- front left leg (coxa, femur, tibia)
+                              gait_direction * -1, -1,  1,    -- back left leg (coxa, femur, tibia)
+                              gait_direction * -1,  1, -1}    -- back right leg (coxa, femur, tibia)
 
     if arming:is_armed() then
         angles = main_inverse_kinematics()
```

Comment:
This should be a local inside the function (maybe declared above this if block), as it's doing a global lookup which is slower, and not needed.

Reply:
did you mean gait_direction should be local ?

Reply:
`angles` looks to only be used right in this local block